### PR TITLE
[1.4.x] Add support for the handle service generation when requestBody  has `oneOf` schema type

### PR DESCRIPTION
**Pull request by lnash94**
```openapi
  /pet:
    put:
      summary: "scenario 01: Request body is with oneOf schema for application/json"
      operationId: putKey
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: "#/components/schemas/Dog"
                - $ref: "#/components/schemas/Cat"
      responses:
        200:
          description: Successful operation.
```
Generated service
```ballerina
public type Cat record {|
    string name?;
    string kind?;
|};

public type Dog record {|
    string name?;
    string age?;
|};

public type Pet_body Dog|Cat;

resource function put pet(@http:Payload Pet_body payload) returns http:Ok {
    }
```
Note: Here `oneOf` schema has been sent under the component section with the name `Pet_body` in OAS by the swagger parser itself.

## Automation tests
 - Unit tests - added
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java [hunk 1]
```diff
@@ -275,6 +264,26 @@ public static ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitio
         }
     }
 
+    /**
+     *
+     * This util uses for selecting standard media type by looking at the user defined media type.
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java:269]:
```suggestion
     * This util is used for selecting standard media type by looking at the user defined media type.
```

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java [hunk 2]
```diff
@@ -275,6 +264,26 @@ public static ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitio
...
+            mediaTypeContent = GeneratorConstants.APPLICATION_JSON;
+        } else if (mediaTypeContent.matches("application/.*\\+xml")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_XML;
+        } else if (mediaTypeContent.matches("application/.*\\+octet-stream")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_OCTET_STREAM;
+        } else if (mediaTypeContent.matches("application/.*\\+x-www-form-urlencoded")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_URL_ENCODE;
+        }
+        return mediaTypeContent;
+    }
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java:285]: whats the expected output for the media types like `text/json`, `text/xml`, etc?

Comment (lnash94) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java:285]: Currently, the tool was mapping with the string type. will re-change the util as per discussed in the client generation media type mapping. `text/json ` for JSON and `text/xml` for XML.

File: openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java [hunk 3]
```diff
@@ -610,6 +612,13 @@ public void escapeIdentifierTest() {
         Assert.assertEquals(GeneratorUtils.escapeIdentifier("foo_bar$_baz"), "'foo_bar\\$_baz");
     }
 
+    @Test
+    public void testForSelectMediaType() {
+        Assert.assertEquals(ServiceGenerationUtils.selectMediaType("text/json"),
+                GeneratorConstants.APPLICATION_JSON);
+        Assert.assertEquals(ServiceGenerationUtils.selectMediaType("text/xml"),
+                GeneratorConstants.APPLICATION_XML);
+    }
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java:621]: missing blank line

**Review by NipunaRanasinghe — APPROVED** [review]

**Comment by lnash94** [comment]
Please note: will merge after the Ballerina 2201.4.0 release.

**Pull request merged**